### PR TITLE
[LFRic] Remove support for old-style `arg_type` descriptor (for #874)

### PR DESCRIPTION
**Pull request by TeranIvy**
As mentioned in #874, this is part of the preparation for the PSyclone 2.0.0 release.

### CODE REVIEW

File: src/psyclone/tests/test_files/dynamo0p3/infrastructure/argument_mod.F90 [hunk 1]
```diff
@@ -213,11 +213,7 @@ module argument_mod
 
   !> @defgroup operates_on Enumeration of kernel iterator property descriptors.
   !> @{
-  ! Note: `iterates_over = CELLS` changes to `operates_on = CELL_COLUMN` in
-  !       PSyclone 2.0.0. Currently both options are supported for uninterrupted
-  !       development, however the `CELLS` option will be removed in #870 or #874).
   integer, public, parameter :: CELL_COLUMN = 396
```
Comment (arporter) [inline src/psyclone/tests/test_files/dynamo0p3/infrastructure/argument_mod.F90:216]: I would have expected to see DOF in here but I've grep'd and it isn't. However, the built-in definitions file (https://github.com/stfc/PSyclone/blob/f33db525f522ef5cbb4263093b4706bb119fc1a9/src/psyclone/parse/lfric_builtins_mod.f90#L48-L53) needs it.

File: src/psyclone/tests/test_files/dynamo0p3/infrastructure/argument_mod.F90 [hunk 2]
```diff
@@ -226,22 +222,19 @@ module argument_mod
...
-     !! GH_FIELD, GH_OPERATOR, GH_COLUMNWISE_OPERATOR}. One of
-     !! {GH_SCALAR, GH_FIELD, GH_OPERATOR, GH_COLUMNWISE_OPERATOR}.
+     !! column-wise operator). One of {GH_SCALAR, GH_FIELD, GH_OPERATOR,
+     !! GH_COLUMNWISE_OPERATOR}.
      integer :: argument_type
-     !> @todo in #874: Add Fortran primitive type of a kernel argument data
-     !!                One of {GH_REAL, GH_INTEGER} (note that these values
-     !!                are currently used to denote scalar argument types).
-     ! integer :: data_type ! To be enabled in #874
+     !> Fortran primitive type of a kernel argument data.
```
Comment (arporter) [inline src/psyclone/tests/test_files/dynamo0p3/infrastructure/argument_mod.F90:235]: s/of a/of/

**Review by arporter — CHANGES_REQUESTED** [review]
Almost perfect. I think think it demonstrates that you did a good job when adding all those TODOs for 874.
All tests and examples are OK with compilation.
Documentation all builds cleanly and does not require updating.
No mention of 874 in any Python or documentation files.
The only thing is that I noticed that argument_mod does not contain a definition of `DOF` and it should (the Fortran file defining the metadata for the built-ins requires it).

File: src/psyclone/tests/test_files/dynamo0p3/infrastructure/argument_mod.F90 (hunk 1 above)
Comment (TeranIvy) [reply src/psyclone/tests/test_files/dynamo0p3/infrastructure/argument_mod.F90:216]: That's strange, the built-ins test with compilation worked. I renamed the former `CELLS` id to `DOF`.

File: src/psyclone/tests/test_files/dynamo0p3/infrastructure/argument_mod.F90 (hunk 2 above)
Comment (TeranIvy) [reply src/psyclone/tests/test_files/dynamo0p3/infrastructure/argument_mod.F90:235]: Thanks, done!

**Comment by TeranIvy** [comment]
Tests are fine so back to @arporter for another look.

File: src/psyclone/tests/test_files/dynamo0p3/infrastructure/argument_mod.F90 (hunk 1 above)
Comment (arporter) [reply src/psyclone/tests/test_files/dynamo0p3/infrastructure/argument_mod.F90:216]: We never compile this file - it's just used to provide the kernel metadata.

**Review by arporter — APPROVED** [review]
All requested changes have been made.
Will proceed to merge.

**Pull request merged**